### ISSUE TITLE
MobileNet models for Text classification

### ISSUE THREAD

**Issue by loretoparisi** [tensorflow/models#2972]
Given the new tensorflow lite and new [upcoming](https://techcrunch.com/2017/12/06/qualcomms-snapdragon-845-focuses-on-ai-vr-and-battery-life/) mobile cpu hardware support for it, it would great to add pre-trained [MobileNet](https://github.com/tensorflow/models/blob/master/research/slim/nets/mobilenet_v1.md) models for binary (spam,ham) or multi-label Text Classification (CNN or Word2Vec), Sentences Embedding (Word2Vec), RNN Sentiment Classification, etc.
Also more tutorial how to train a MobileNet like model for text based tasks would be helpful.

**Comment by nguyeho7** [comment]
MobileNets are convolutional network architectures for image classification, IIRC. You can of course make your own small recurrent or fully connected network for those tasks and push them onto your phone, but I'd say Word2Vec by itself is already too large to fit into most phone memory.

**Comment by loretoparisi** [comment]
@nguyeho7 thanks for the clarification, this is a very challenging task. In fact regarding Word2Vec, Mikolov et al. used a [quantization](https://arxiv.org/abs/1612.03651) approach for Word2Vec embedding models, that should fit at runtime in recent phone.
By example the FastText language identification model (172 languages) features a 98.5% of accuracy with the standard model (subwords embedding) that is 350MB, and less than 1MB model with an acceptable accuracy. See [here](https://fasttext.cc/blog/2017/10/02/blog-post.html).
Following this approach, could be interesting to create MobileNet models for text based tasks then as well.

**Comment by drpngx** [comment]
More examples sounds like a good idea.

/CC @nealwu @tfboyd

**Comment by nealwu** [comment]
FYI in case you haven't already seen it, we do have a MobileNet implementation [here](https://github.com/tensorflow/models/tree/master/research/slim) (search the page for "MobileNet").

**Comment by loretoparisi** [comment]
@nealwu thanks! It would be awesome to have in TF-slim text based tasks, since it is not so easy to translate/port the image based cnn tasks into rnn/cnn based classification tasks to run on mobilenet.

**Comment by suharshs** [comment]
Thanks for the suggestions on pretrained models to release. We are prioritizing various models to train as we get time. Will close this for now.
